pkg/vsphere/extraconfig/keys.go

```diff
@@ -161,7 +161,7 @@ func calculateScopeFromKey(key string) []string {
 	return scopes
 }
 
-func calculateKeyFromField(field reflect.StructField, prefix string, depth recursion) (string, recursion) {
+func calculateKeyFromField(log logrus.FieldLogger, field reflect.StructField, prefix string, depth recursion) (string, recursion) {
```

Comment:
You could just have one logger in the package and use that here. This looks very strange passing in a logger to a function. I would be ok using `trace.Operation` here. But I think using just one package level logger would be much simpler for now.